custom-metrics-stackdriver-adapter/pkg/adapter/provider/query_builder.go

```diff
@@ -67,6 +65,18 @@ type Translator struct {
 	useNewResourceModel bool
 }
 
+func NewTranslator(service *stackdriver.Service, gceConf *config.GceConfig, rateInterval time.Duration, alignmentPeriod time.Duration, c clock, mapper apimeta.RESTMapper, useNewResourceModel bool) *Translator {
+	return &Translator{
+		service:             service,
+		config:              gceConf,
+		reqWindow:           rateInterval,
+		alignmentPeriod:     alignmentPeriod,
+		clock:               c,
```

Comment:
you dont need 'c' arg in this methos
simply assign a real clock to clock fields
kin tests you can replace   clock field with fake clock

what is the rationale for this change? Maybe I don't get something.